### ISSUE TITLE
State Wise Vaccination Coverage

### ISSUE THREAD

**Issue by thejeshgn** [datameet/covid19#31]
I can see that mohfw is adding State Wise Vaccination Coverage.

https://www.mohfw.gov.in/pdf/CumulativeCOVIDVaccinationCoverage7thMarch2021.pdf

**Comment by thejeshgn** [comment]
File name is misleading. It sas 7th March 2021. Inside it says  `As on 08 Mar’21 at 7:00 AM`. We are going to name the files as per this datetime. Like `2021-03-08-at-07-00-AM.pdf`
`

**Comment by naveenpf** [comment]
Vaccination coverage by statewise
https://dashboard.cowin.gov.in/ 

Didn't see stats API
https://apisetu.gov.in/public/marketplace/api/cowin

**Comment by thejeshgn** [comment]
@naveenpf it looks a standard dashboard to give overall stats. I couldn't get the previous data or data on a specific date outside last 30 days. "All" has weekly data.

**Comment by sajinh** [comment]
If it helps, state wise vaccination data can also be found as JSON document from mygov. Here is how I get them

curl https://www.mygov.in/sites/default/files/covid/vaccine/covid_vaccine_timeline.json?#{itim} -o ./vdata.#{stim}`

time = Time.now
itim = time.to_i
stim = time.strftime("%m-%d-%Y")

**Comment by thejeshgn** [comment]
does it match with the PDF on MOHFW, which they publish everyday.

https://www.mohfw.gov.in/pdf/CumulativeCOVIDVaccinationCoverageReport06052021.pdf



Thej
--
Thejesh GN *⏚* ತೇಜೇಶ್ ಜಿ.ಎನ್
http://thejeshgn.com
GPG ID :  0xBFFC8DD3C06DD6B0

**Comment by sajinh** [comment]
Yes, it matches except for rows 8--9 ("Dadra & Nagar Haveli" and "Daman Diu"). In the JSON file these are lumped together into one.  

The data that I have collected and processed (sorted into states) for my own analysis can be found here:
https://framagit.org/saji_nh/india_2020

**Comment by bkamapantula** [comment]
hi @thejeshgn, I'm curious if anyone has made progress on this.. a friend is looking for vaccine doses by gender. I thought I'll check here and see if anyone has already scraped it.

**Comment by thejeshgn** [comment]
@bkamapantula I have not seen any dataset with gender wise details. Have you?

**Comment by thejeshgn** [comment]
@sajinh thank you. I will add it to the auto scraper.

**Comment by bkamapantula** [comment]
@thejeshgn not quite the API but here's the request for Arunchal Pradesh, Changlang district (picked from the Network tab on Cowin dashboard site). has to be scraped out

https://api.cowin.gov.in/api/v1/reports/v2/getPublicReports?state_id=3&district_id=20&date=2021-05-16

at gender level this is: vaccination numbers by male/female/other at state, district granularity

**Comment by bkamapantula** [comment]
I scraped it, will share it in a few days.

**Comment by thejeshgn** [comment]
@bkamapantula that will be great.

**Comment by bkamapantula** [comment]
hi @thejeshgn, here it is: [cowin-vaccination-doses](https://docs.google.com/spreadsheets/d/11R2o6Fknvcitbm-zw2SmI34UugsfQm9qJXZUTAKNiUo/edit?usp=sharing)

I update it once in two days.

**Comment by thejeshgn** [comment]
@bkamapantula  So shall I script to download once every other day?

I would like to add it to the repo.

**Comment by bkamapantula** [comment]
Yes, that works @thejeshgn 

Let me know if you need a specific sheet URL, I can update that sheet with new data. 

Currently, I create a new sheet with date as name.

**Comment by shuklaayush** [comment]
@thejeshgn Looks like the 15th March bulletin was [revised](https://www.mohfw.gov.in/pdf/CumulativeCOVIDVaccinationReport15thMarch2021.pdf) and the repository still has the [old one](https://github.com/datameet/covid19/blob/master/downloads/mohfw-backup/cumulative_vaccination_coverage/2021-03-16-at-07-00-AM.pdf). 

I also found a few more files that maybe you could add. The filenames are a mess
https://www.mohfw.gov.in/pdf/CumulativeCovidVaccinationCoverageReportDay40ason25Febat7AM.pdf
https://www.mohfw.gov.in/pdf/CumulativeCovidVaccinationCoverageReportDay41ason26thFebat7am.pdf
https://www.mohfw.gov.in/pdf/CumulativeCovidVaccinationCoverageReportDay42ason27thFebat7am.pdf
https://www.mohfw.gov.in/pdf/CumulativeCovidVaccinationCoverageReportDay43ason28thFebat7am.pdf
https://www.mohfw.gov.in/pdf/CumulativeCoverageReport2ndMarch2021.pdf (1st March)
https://www.mohfw.gov.in/pdf/CumulativeCOVIDVaccinationCoverageReport2March2021.pdf
https://www.mohfw.gov.in/pdf/CumulativeCoverageReport3rdMarch2021.pdf
https://www.mohfw.gov.in/pdf/CumulativeCoverageReport4thMarch2021.pdf
https://www.mohfw.gov.in/pdf/CumulativeCOVIDVaccinationCoverageReport5thMarch2021.pdf
https://www.mohfw.gov.in/pdf/CumulativeCOVIDVaccinationReport6thMarch2021.pdf

**Comment by thejeshgn** [comment]
@shuklaayush thank you. Will look and update tonight.

**Comment by thejeshgn** [comment]
@shuklaayush I usually use the datetime  inside the PDF `(As on 16 Mar’21 at 7:00 AM)` as the file name to keep it consistent.

**Comment by thejeshgn** [comment]
@shuklaayush added to the repo.

**Comment by thejeshgn** [comment]
@bkamapantula is there is a possibility of committing the code ? That will help the reproducibility.

**Comment by bkamapantula** [comment]
sure, here it is: https://gitlab.com/bhanu.k/cowin-vaccination-doses

The script isn't auto-setup deliberately to monitor for failed requests.

**Comment by bkamapantula** [comment]
@thejeshgn I cleaned up the [instructions](https://gitlab.com/bhanu.k/cowin-vaccination-doses#steps-to-get-data) and updated code with better control for arguments.

Data files are available via Git LFS.

**Comment by thejeshgn** [comment]
@bkamapantula I have not gotten around to work on it. But its on my list. I will get to it this weekend. Thank you so much.
